changelog.d/10667.misc

```diff
@@ -0,0 +1 @@
+Flatten the `tests.synapse.rests` package by moving the contents of `v1` and `v2_alpha` into the parent.
```

Comment:
Could use the same changelog as #10600 if you wanted to combine them, but they're both `misc` so probably doesn't matter.

Reply:
What's the mechanism there, sorry? Will towncrier spot that two changelogs have the same contents and report them only once?

Reply:
> What's the mechanism there, sorry? Will towncrier spot that two changelogs have the same contents and report them only once?

Yes, it will combine them onto a single line and link to all the PRs. Something like "My awesome change. (123, 456)"